recipes/opentelemetry-cpp/all/conanfile.py

```diff
@@ -408,6 +434,9 @@ def package_info(self):
         if self.settings.os in ("Linux", "FreeBSD"):
             self.cpp_info.components["opentelemetry_common"].system_libs.extend(["pthread"])
 
+        if Version(self.version) >= "1.16.0" and is_apple_os(self):
+            self.cpp_info.components["opentelemetry_common"].frameworks.extend(["CoreFoundation"])
```

Comment:
```suggestion
            self.cpp_info.components["opentelemetry_common"].frameworks.extend(["CoreFoundation"])
self.cpp_info.components["opentelemetry_common"].defines.append("ENABLE_OTLP_COMPRESSION_PREVIEW")
```

https://github.com/open-telemetry/opentelemetry-cpp/compare/v1.14.2...v1.16.1#diff-1186bce31e69b2154d84fff38d88b12365327134f5b3ed31e734889784a948dbR135

https://github.com/open-telemetry/opentelemetry-cpp/blob/a713947e5956bbc4b5a398b148e497ce21278e69/ext/src/http/client/curl/http_client_curl.cc#L26

Reply:
Just a note: that the define `ENABLE_OTLP_COMPRESSION_PREVIEW` doesn't depends of `is_apple_os` condition , only the -framework `CoreFoundation`:
https://github.com/open-telemetry/opentelemetry-cpp/blob/a713947e5956bbc4b5a398b148e497ce21278e69/api/CMakeLists.txt#L129

Reply:
Great catch!